### ISSUE TITLE
[Phase 4] Feature Extraction

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/neuronal-spike-sorting#5]
## Objective

Implement PCA-based feature extraction from spike waveforms.

## Dependencies

- Phase 3: Spike Detection (requires extracted waveforms)

## Tasks

- [ ] Implement `src/features/pca_features.py` for PCA extraction
- [ ] Implement `src/features/waveform_features.py` for manual features
- [ ] Determine optimal number of PCA components
- [ ] Handle multi-channel data appropriately
- [ ] Normalize features for clustering
- [ ] Save features to `data/features/spike_features.npz`
- [ ] Write unit tests

## Files to Create

| File | Purpose |
|------|---------|
| `src/features/__init__.py` | Subpackage init |
| `src/features/pca_features.py` | PCA on waveforms |
| `src/features/waveform_features.py` | Manual features |
| `tests/test_features.py` | Feature tests |

## Starter Code

```python
# src/features/pca_features.py
"""PCA feature extraction for spike waveforms."""

import numpy as np
from sklearn.decomposition import PCA
from sklearn.preprocessing import StandardScaler

class WaveformPCA:
    """PCA feature extractor for spike waveforms."""

    def __init__(self, n_components: int = 10):
        self.n_components = n_components
        self.pca = PCA(n_components=n_components)
        self.scaler = StandardScaler()

    def fit(self, waveforms: np.ndarray) -> "WaveformPCA":
        """
        Fit PCA on waveforms.

        Parameters
        ----------
        waveforms : np.ndarray
            Spike waveforms, shape (n_spikes, n_samples) or
            (n_spikes, n_channels, n_samples)
        """
        # Flatten multi-channel waveforms
        if waveforms.ndim == 3:
            n_spikes = waveforms.shape[0]
            waveforms = waveforms.reshape(n_spikes, -1)

        # Scale before PCA
        waveforms_scaled = self.scaler.fit_transform(waveforms)
        self.pca.fit(waveforms_scaled)
        return self

    def transform(self, waveforms: np.ndarray) -> np.ndarray:
        """Extract PCA features."""
        if waveforms.ndim == 3:
            n_spikes = waveforms.shape[0]
            waveforms = waveforms.reshape(n_spikes, -1)

        waveforms_scaled = self.scaler.transform(waveforms)
        return self.pca.transform(waveforms_scaled)

    def fit_transform(self, waveforms: np.ndarray) -> np.ndarray:
        """Fit and transform in one step."""
        self.fit(waveforms)
        return self.transform(waveforms)

    def explained_variance_ratio(self) -> np.ndarray:
        """Return explained variance ratio."""
        return self.pca.explained_variance_ratio_
```

## Definition of Done

- [ ] PCA reduces waveforms to 10 features
- [ ] Features visualizable in 2D projection
- [ ] Multi-channel handling correct
- [ ] Explained variance > 90% with 10 components
- [ ] Feature matrix saved
- [ ] All tests passing

## Technical Notes

- Flatten multi-channel waveforms before PCA
- Scale features before PCA for better convergence
- 10 components typically capture >90% variance

---
Part of #1 (Meta Issue)
